### PR TITLE
WiP: filename convention for bedfiles

### PR DESCRIPTION
**Pull request by hassanfa**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
 * [Maintainers](#maintainers)
 * [Versioning Schema](#versioning)
 
+#### Filename convention:
+
+name\_{project_number.version}\_genome\_application/purpose.{bedgraph,bed,cnn,bigwig,vcf}
```
Comment (henrikstranneheim) [inline README.md]:
```suggestion
{name}_{project_number.version}_{genome}_{application/purpose}.{bedgraph,bed,cnn,bigwig,vcf}
```
Or what was the meaning of the escape character "\"

File: README.md [hunk 2]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+example1: GMCKsolid\_4.1\_hg19\_varcall.bed
```
Comment (henrikstranneheim) [inline README.md]: This looks like a path to me and not a file name - why not:
```suggestion
example1: GMCKsolid_4.1_hg19_varcall.bed
```

File: README.md [hunk 3]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+example2: GMCKsolid\_4.1\_hg19\_pon.cnn
+
+example3: TwistExome\_???.???\_hg19\_design.bed
```
Comment (henrikstranneheim) [inline README.md]: The question marks make me reallt confused. Am I suppose to add "?" if I do not have a version or is it just to show that this version is not set? Either way I would avoid "?" in file names.

File: README.md [hunk 4]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+example4: TwistExomeRefSeq\_???.???\_hg19\_design.bed
+
+name:
+
+- ....
```
Comment (henrikstranneheim) [inline README.md]: What does this mean?

File: README.md [hunk 5]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+project_number:
+
+- 1,2,3,4,...
```
Comment (henrikstranneheim) [inline README.md]: Should this follow semantiv versioning or is thera a definition on the versioning format expect that it is an INT? I am not up to speed on this - Is the project number relevant in this regard? Enough to put it in the filename? I guess it traces how the file was created or?

File: README.md [hunk 6]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+version:
+
+- Any change to bait or content
```
Comment (henrikstranneheim) [inline README.md]: Is it not enough to say any change to file content?

File: README.md [hunk 7]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+name:
+
+- ....
+
+project_number:
+
+- 1,2,3,4,...
+
+version:
```
Comment (henrikstranneheim) [inline README.md]:
```suggestion
file_version:
```
or?

File: README.md [hunk 8]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+- Any change to bait or content
+- new product from twist
+- TBA
```
Comment (henrikstranneheim) [inline README.md]: What is TBA? To be announced?

File: README.md [hunk 9]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+
+genome:
+
+- hg19, hg38
+
+application/purpose:
+
+1. target: original target file that was sent to Twist
```
Comment (henrikstranneheim) [inline README.md]: Very nice!

File: README.md [hunk 10]
```diff
@@ -12,6 +12,54 @@ Repository to collect and share bed files for each bait set. This repository doe
...
+2. design: all\_target\_segments\_covered\_by\_probes file received from Twist
+3. dropout: all\_target\_segments\_**not**\_covered\_by\_probes file received from Twist
+3. varcall: file is suitable for variantcalling based on .....
+4. ucsctrack: file is suitable to upload to ucsc as track file ....
+5. pon: file is a pool/panel of normals
+6. TBA
+
+extension:
```
Comment (henrikstranneheim) [inline README.md]: This is great but is not file_suffix or file_ending or even "format" more appropriate?

File: README.md (hunk 3 above)
Comment (hassanfa) [reply README.md]: This is still work in progress. Version, project number and purpose for twist was still Vague. Anna Ly and Anna G have come up with new ids and I will update this as we go.

File: README.md (hunk 4 above)
Comment (hassanfa) [reply README.md]: Same as above, still not sure about the actual name. Open to suggestions and changes. Discussion also ongoing with Anna Ly.

File: README.md (hunk 5 above)
Comment (hassanfa) [reply README.md]: This is not semantic versioning. Twist kits are versioned by project followed by dot and an integer showing version. Any change in content of bed file changes the version.

This number ultimately ends up on tubes in lab, and thus it is easier to remember and write it down.

For example in 4.1: 4 will always be GMCKSolid's, and 1 will be the version.

File: README.md (hunk 6 above)
Comment (hassanfa) [reply README.md]: Change in the header will not be reflected in version. This is to make labeling in lab easier.

File: README.md (hunk 8 above)
Comment (hassanfa) [reply README.md]: yes :-) Discussion ongoing with Anna Ly. We have couple of ideas to add here.

File: README.md (hunk 10 above)
Comment (hassanfa) [reply README.md]: I'm skeptical if we should even keep these here. The idea is to keep track of what happened to the original bed file. But those files will end up in produciton/{balsamic,mip}/ path.

The whole purpose of this repo is to facilitate lab and bioinfo coordination, that's the reason to adding extensions, purpose, etc.

Comment (hassanfa) [reply README.md]: Aslo, I agree. "format" is more suitable, or even "file_extension".

File: README.md (hunk 2 above)
Comment (hassanfa) [reply README.md]: This is for .md only, it shows up as italic otherwise

**Comment by hassanfa** [comment]
Proof that new files are same as old files:

|shasum_algo_1|file_name|
|---|---|
|122f8a2036732e43d5fe379d170f5ca62fc6f018|data_files/LymphoMATIC2.1_hg19.bed|
|122f8a2036732e43d5fe379d170f5ca62fc6f018|data_files/lymphomatic_2.1_hg19_design.bed|
|1403805685526852d4536dfeaea0d364043fe2e6|data_files/GIcfDNA3.1_hg19.bed|
|1403805685526852d4536dfeaea0d364043fe2e6|data_files/gicfdna_3.1_hg19_design.bed|
|404075f1549bae7c974debbccc740bf3a0669e22|data_files/GMSmyeloid5.2_hg19.bed|
|404075f1549bae7c974debbccc740bf3a0669e22|data_files/gmsmyeloid_5.2_hg19_design.bed|
|4dd8656f0f23d3e904322a35ec867ee667cc1584|data_files/Twist_Target_hg19_whole_exome.bed|
|4dd8656f0f23d3e904322a35ec867ee667cc1584|data_files/twistexome_8.1_hg19_design.bed|
|5228edaa36f4d3608e33828ff14b55696bcaa2fa|data_files/Lymphoma_hg19.bed|
|5228edaa36f4d3608e33828ff14b55696bcaa2fa|data_files/lymphoma_6.1_hg19_design.bed|
|86ff39b156a2aac40a697b18a8739740c9649fbf|data_files/Twist_Target_hg19_RefSeq_1_22_X_Y.bed|
|86ff39b156a2aac40a697b18a8739740c9649fbf|data_files/twistexomerefseq_9.1_hg19_design.bed|
|8ad734e29865e833f2e048b66efa73d238706814|data_files/GMSLymphoid7.1_hg19.bed|
|8ad734e29865e833f2e048b66efa73d238706814|data_files/gmslymphoid_7.1_hg19_design.bed|
|ab6fbf6e883b4cea409a5aa78248c756901e7cb9|data_files/GMSmyeloid5.1_hg19.bed|
|ab6fbf6e883b4cea409a5aa78248c756901e7cb9|data_files/gmsmyeloid_5.1_hg19_design.bed|
|def7db1ee36543f2dfc5f6d3dcaa4436fff50350|data_files/GMCKSolid4.1_hg19.bed|
|def7db1ee36543f2dfc5f6d3dcaa4436fff50350|data_files/gmcksolid_4.1_hg19_design.bed|

**Comment by hassanfa** [comment]
@annaengstrom ready! :-) Let's merge this and I'll update BALSAMIC with correct reference as well.

**Comment by hassanfa** [comment]
No particular reason, justed to make them all look similar. Maybe I should change them back.

For gicfdna, it wasn't clear if second letter is L or capital i.

**Comment by henrikstranneheim** [comment]
Consider using snake_case i.e. "blue_car" instead of BlueCar if you want to improve readability. From a coding view point I would really appreciate all words being lower case. Then I do not have to remember if it was "GIcfDNA" or "GICfDNA" etc

**Comment by hassanfa** [comment]
Anna, https://github.com/Clinical-Genomics/target_capture_bed/blob/file_name_convention/README.md short names changed and if you ok it for lims, I'll go ahead and merge this PR. Then we followup over CGweb/CG to make necessary changes.

**Comment by annaengstrom** [comment]
> Anna, https://github.com/Clinical-Genomics/target_capture_bed/blob/file_name_convention/README.md short names changed and if you ok it for lims, I'll go ahead and merge this PR. Then we followup over CGweb/CG to make necessary changes.

Looks perfect!

**Pull request merged**